functions/src/scripts/readInput.ts

```diff
@@ -0,0 +1,22 @@
+import * as yargs from 'yargs';
+const args = yargs.option('inputPath', { alias: 'i', demand: true, type: 'string' }).argv;
+
+export function readInputFile(): any {
+    const fs = require('fs');
+    const rawJSON = fs.readFileSync(args.inputPath);
+    return rawJSON;
+}
+
+export function getVersion(): any {
+    const splitString = args.inputPath.split('-');
+    let version: string | number | null = null;
+    let versionArray = null;
+    if (splitString.length === 6) { //assuming format of json file name is <year>-<month>-partner-bulletin-preview<-version>.json
+        versionArray = splitString[5].split('.')[0].split('v');
+        version = versionArray[1];
+    } else {
+        version = 1.0;
+    }
```

Comment:
This code does not handle some edge cases. What if the version part does not conform to the format expected by line 15. What if we have 7 parts, which probably would include version, but with some additional data? The logic here should be:
splitString.length == 5 => 1.0
splitString.length == 6 => parse version; if parsing error, fail
otherwise fail